src/sentry/api/endpoints/user_details.py

```diff
@@ -221,15 +217,31 @@ def delete(self, request, user):
         if remaining_org_ids:
             OrganizationMember.objects.filter(
                 organization__in=remaining_org_ids,
-                user=request.user,
+                user=user,
             ).delete()
 
-        User.objects.filter(
-            id=request.user.id,
-        ).update(
-            is_active=False,
-        )
+        logging_data = {
+            'actor_id': request.user.id,
+            'ip_address': request.META['REMOTE_ADDR'],
+        }
+
+        hard_delete = serializer.validated_data.get('hardDelete', False)
+
+        # Only active superusers can hard delete accounts
+        if hard_delete and not is_active_superuser(request):
+            return Response(
+                {'detail': 'Only superusers may hard delete a user account'},
+                status=status.HTTP_400_BAD_REQUEST)
+
+        if hard_delete:
+            user.delete()
+            delete_logger.info('user.removed', extra=logging_data)
+        else:
+            User.objects.filter(id=user.id).update(is_active=False)
+            delete_logger.info('user.deactivate', extra=logging_data)
 
-        logout(request)
+        # if the user deleted their own account log them out
+        if request.user.id == user.id:
```

Comment:
This won't work when the user is hard deleted, because django sets the `id` column to `None` on delete.

Reply:
Cool, I'll just do this check earlier in that case.